### PR TITLE
feat:support user feedback, promote management, etc.

### PR DESCRIPTION
**Pull request by Armandwt**
Submit new features,
1、support user feedback, 
2、promote management, 
3、Add information tips to the knowledge selector.
4、Display the source of knowledge base files.
etc.

### CODE REVIEW

File: components/prompt/prompt-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+          })
+          message.success('编辑成功')
+          setLoading(false)
+          setShow(false)
+        } else {
+          setLoading(false)
+          message.error(res?.err_msg)
+        }
+      } else { // 新增
+        const res: any = await sendSpacePostRequest('/prompt/add', params)
```
Comment (2976151305) [inline components/prompt/prompt-modal.tsx:48]: [api specification](https://www.yuque.com/eosphoros/dbgpt/bffhffkd8pcg3afw#e13d9238)

File: pages/index.tsx
Comment (2976151305) [inline pages/index.tsx:1]: The file will be refactored in the next version, so no adjustments are needed.

File: components/common/prompt-bot.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  const userInfo: any = useUserInfo()
+  const [open, setOpen] = useState(false)
+  const [current, setCurrent] = useState('common')
+
+  const { data, loading } = useRequest(() => {
+    const body = {
+      user_name: userInfo.loginName,
+      prompt_type: current,
+    }
+    return sendSpacePostRequest('/prompt/list', body)
```
Comment (2976151305) [inline components/common/prompt-bot.tsx:61]: [api specification](https://www.yuque.com/eosphoros/dbgpt/bffhffkd8pcg3afw#e13d9238)

File: components/chat/completion.tsx [hunk 3]
```diff
@@ -159,24 +200,43 @@ const Completion = ({ messages, onSubmit, paramsObj = {}, clearInitMessage }: Pr
...
+                    <Option key={item.key} value={item.value}>
+                      {item.key}
+                    </Option>
+                  </AntdTooltip>
                 ))}
               </Select>
             </div>
           )}
-          <Input
+          <Textarea
```
Comment (2976151305) [inline components/chat/completion.tsx:212]: The conversation input box has been refactored in the next version, so there is no need to update it here.

File: pages/prompt/index.tsx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    },
+  })
+
+  const { data, loading, run, mutate } = useRequest((params = tableParams, prompt_type = current) => {
+    const body = {
+      user_name: userInfo.loginName,
+      prompt_type,
+      ...params.pagination,
+    }
+    return sendSpacePostRequest('/prompt/list', body)
```
[the remark this answers is not in the document]
Comment (Armandwt) [reply pages/prompt/index.tsx:85]: It is maybe can't modify according to the development rules. Its situation is the same as Knowledge, without the prefix of api/v1.

File: components/prompt/prompt-modal.tsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  };
+
+  const submit = async () => {
+    try {
+      const values = form.getFieldsValue();
+      const params = { prompt_type, ...values };
+      setLoading(true);
+      if (item) {
+        // edit
+        const res: any = await sendSpacePostRequest('/prompt/update', params);
```
Comment (Aralhi) [inline components/prompt/prompt-modal.tsx:32]: api request should be defined in `client/api/request.ts`.

**Review by Aralhi — APPROVED** [review]

**Pull request merged**